Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift

```diff
@@ -292,7 +292,21 @@ extension BrowserViewController: URLBarDelegate {
          revert `searchURL` from `var` to `let`
          when the Bing experiment will be removed
          */
-        if var searchURL = engine.searchURLForQuery(text) {            
+        if var searchURL = engine.searchURLForQuery(text) {         
```

Comment:
Also realised, given we have the same logic of constructing the url depending on the variant in two places, how about moving it inside `engine.searchURLForQuery(...)` since this seems to be common? Or do you see any other issue with doing that?

Reply:
I'd favor a single place as you suggested as well 👍 
THoug, If we add it as part of `engine.searchURLForQuery(...)` then features like the Search Engine Shortcuts (e.g.  `didSelectEngine` event) would make it hard to differentiate. I know there is a high change that both features won't be enabled at the same time in the same regions but I liked not to count on this condition and make it both independent